src/flows/OnboardingFlow/steps/AddUserInfo.tsx

```diff
@@ -25,51 +25,36 @@ function useWizardConfig({ onNext }: ConfigProps) {
 }
 
 function AddUserInfo({ next }: WizardContext) {
-  // TODO(Terence): Investigate using a preloaded query here to ensure the user
-  // never gets a loading state when navigating to the next step.
-  const { viewer } = useLazyLoadQuery<AddUserInfoQuery>(
-    graphql`
-      query AddUserInfoQuery {
-        viewer {
-          ... on Viewer {
-            user {
-              dbid
-              bio
-              username
-            }
-          }
-        }
-      }
-    `,
-    {}
-  );
+  const { username, onUsernameChange, usernameError, bio, onBioChange, generalError, onEditUser } =
+    useUserInfoForm({
+      onSuccess: next,
+      userId: undefined,
+      existingUsername: '',
+      existingBio: '',
+    });
 
-  if (!viewer?.user) {
-    throw new Error('Entered the AddUserInfo step without a logged in user in the cache');
-  }
+  const { setNextEnabled } = useWizardValidationActions();
 
-  const {
-    username,
-    onUsernameChange,
-    usernameError,
-    onClearUsernameError,
-    bio,
-    onBioChange,
-    generalError,
-    onEditUser,
-  } = useUserInfoForm({
-    onSuccess: next,
-    userId: viewer.user.dbid,
-    existingUsername: viewer.user.username ?? undefined,
-    existingBio: viewer.user.bio ?? undefined,
-  });
+  useEffect(() => {
+    if (username.length < 2) {
+      setNextEnabled(false);
+      return;
+    }
+    setNextEnabled(!usernameError);
+  }, [setNextEnabled, username, usernameError]);
```

Comment:
ensure the "next" button is disabled until the username is confirmed to be valid.

we didn't need this before because we actually needed the user to click the "next" button since errors were displayed _on-submit_, rather than on-type.